packages/mattermost-redux/src/actions/posts.ts

```diff
@@ -1242,3 +1246,21 @@ export function moveHistoryIndexForward(index: number) {
         return {data: true};
     };
 }
+
+/**
+ * Ensures thread-replies in channels correctly follow CRT:ON/OFF
+ */
+export function resetReloadPostsInChannel() {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        dispatch({
+            type: PostTypes.RESET_POSTS_IN_CHANNEL,
+        });
+
+        const currentChannelId = getCurrentChannelId(getState());
+        if (currentChannelId) {
+            await dispatch(selectChannel('')); // this await is needed, do not remove
```

Comment:
Could you elaborate as to why we need this?

Reply:
Updated the comment. In short it's needed to ensure the posts reload after the re-select (otherwise it will be stuck on the loading screen)